In [160]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import statsmodels.api as sm
import numpy as np

In [161]:
da = pd.read_csv("winequality-white (1).csv")

In [162]:
da.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.0              0.27         0.36            20.7      0.045   
1            6.3              0.30         0.34             1.6      0.049   
2            8.1              0.28         0.40             6.9      0.050   
3            7.2              0.23         0.32             8.5      0.058   
4            7.2              0.23         0.32             8.5      0.058   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 45.0                 170.0   1.0010  3.00       0.45   
1                 14.0                 132.0   0.9940  3.30       0.49   
2                 30.0                  97.0   0.9951  3.26       0.44   
3                 47.0                 186.0   0.9956  3.19       0.40   
4                 47.0                 186.0   0.9956  3.19       0.40   

   alcohol  quality  
0      8.8        6  
1      9.5        6  
2     10.1        6  
3      9.9        6  
4      9.9        6

In [164]:
da.rename(columns={'fixed acidity': 'fixed_acidity'}, inplace=True)
da.rename(columns={'volatile acidity': 'volatile_acidity'}, inplace=True)
da.rename(columns={'citric acid': 'citric_acid'}, inplace=True)
da.rename(columns={'residual sugar': 'residual_sugar'}, inplace=True)
da.rename(columns={'free sulfur dioxide': 'free_sulfur_dioxide'}, inplace=True)
da.rename(columns={'total sulfur dioxide': 'total_sulfur_dioxide'}, inplace=True)
model = sm.OLS.from_formula("quality ~ fixed_acidity+volatile_acidity+citric_acid+residual_sugar+free_sulfur_dioxide+total_sulfur_dioxide+fixed_acidity+density+chlorides+pH+sulphates+alcohol" , data=da)
result = model.fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                quality   R-squared:                       0.282
Model:                            OLS   Adj. R-squared:                  0.280
Method:                 Least Squares   F-statistic:                     174.3
Date:                Tue, 24 Oct 2023   Prob (F-statistic):               0.00
Time:                        17:43:41   Log-Likelihood:                -5543.7
No. Observations:                4898   AIC:                         1.111e+04
Df Residuals:                    4886   BIC:                         1.119e+04
Df Model:                          11                                         
Covariance Type:            nonrobust                                         
========================================================================================
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
Intercept              150.1928     18.804      7.987      0.000     113.328     187.057
fixed_acidity            0.0655      0.021      3.139      0.002       0.025       0.106
volatile_acidity        -1.8632      0.114    -16.373      0.000      -2.086      -1.640
citric_acid              0.0221      0.096      0.231      0.818      -0.166       0.210
residual_sugar           0.0815      0.008     10.825      0.000       0.067       0.096
free_sulfur_dioxide      0.0037      0.001      4.422      0.000       0.002       0.005
total_sulfur_dioxide    -0.0003      0.000     -0.756      0.450      -0.001       0.000
density               -150.2842     19.075     -7.879      0.000    -187.679    -112.890
chlorides               -0.2473      0.547     -0.452      0.651      -1.319       0.824
pH                       0.6863      0.105      6.513      0.000       0.480       0.893
sulphates                0.6315      0.100      6.291      0.000       0.435       0.828
alcohol                  0.1935      0.024      7.988      0.000       0.146       0.241
==============================================================================
Omnibus:                      114.161   Durbin-Watson:                   1.621
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              251.637
Skew:                           0.073   Prob(JB):                     2.28e-55
Kurtosis:                       4.101   Cond. No.                     3.74e+05
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 3.74e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Çıkarımlarım Şu şekilde:
P>|t| olduğu takdirde değişkenin modeldeki payı anlamlı düzeyde değildir. Citric acid, total sülfür dioksid, chlorides değişkenleri bu lineer regresyon denklemine alınmamalıdır. Diğerleri 0 çıktığından dolayı modele alınması gereken değişkenlerdir. Density kavramının da standart error'u çok yüksek olduğu için modelin doğru tahminlemesini ve kalitesini düşürecektir. Bunun dışında coef başlığı altındaki değerler de denklemde gözükecek katsayılardır.
Bu standart sapması çok büyük ve anlamlılık düzeyi az olan katsayılı değişkenleri çıkardığımda durum şuna dönüşecek:

In [167]:
da.rename(columns={'fixed acidity': 'fixed_acidity'}, inplace=True)
da.rename(columns={'volatile acidity': 'volatile_acidity'}, inplace=True)
da.rename(columns={'citric acid': 'citric_acid'}, inplace=True)
da.rename(columns={'residual sugar': 'residual_sugar'}, inplace=True)
da.rename(columns={'free sulfur dioxide': 'free_sulfur_dioxide'}, inplace=True)
da.rename(columns={'total sulfur dioxide': 'total_sulfur_dioxide'}, inplace=True)
model = sm.OLS.from_formula("quality ~ fixed_acidity+volatile_acidity+residual_sugar+free_sulfur_dioxide+fixed_acidity+pH+sulphates+alcohol" , data=da)
result = model.fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                quality   R-squared:                       0.271
Model:                            OLS   Adj. R-squared:                  0.270
Method:                 Least Squares   F-statistic:                     260.2
Date:                Tue, 24 Oct 2023   Prob (F-statistic):               0.00
Time:                        17:46:42   Log-Likelihood:                -5579.3
No. Observations:                4898   AIC:                         1.117e+04
Df Residuals:                    4890   BIC:                         1.123e+04
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
=======================================================================================
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept               1.8921      0.337      5.611      0.000       1.231       2.553
fixed_acidity          -0.0556      0.014     -3.896      0.000      -0.084      -0.028
volatile_acidity       -2.0344      0.109    -18.683      0.000      -2.248      -1.821
residual_sugar          0.0252      0.003     10.089      0.000       0.020       0.030
free_sulfur_dioxide     0.0036      0.001      5.215      0.000       0.002       0.005
pH                      0.1533      0.081      1.881      0.060      -0.006       0.313
sulphates               0.3832      0.096      3.979      0.000       0.194       0.572
alcohol                 0.3776      0.010     37.650      0.000       0.358       0.397
==============================================================================
Omnibus:                      106.791   Durbin-Watson:                   1.626
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              228.138
Skew:                           0.071   Prob(JB):                     2.89e-50
Kurtosis:                       4.048   Cond. No.                     1.32e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.32e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

# quality= 1.8921-0.0556*fixed_acidity-2.0344*volatile_acidity+0.0252*residual_sugar+0.0036*free_sulfur_dioxide	+0.1533*pH+0.096*sulfat+0.3776*alcohol